x-pack/plugins/cloud_security_posture/server/create_indices/create_processor.ts

```diff
@@ -25,26 +31,8 @@ export const createPipelineIfNotExists = async (
     const exitError = transformError(exitErr);
     if (exitError.statusCode === 404) {
       try {
-        await esClient.ingest.putPipeline({
-          id: pipelineId,
-          description: 'Pipeline for adding event timestamp',
-          processors: [
-            {
-              set: {
-                field: '@timestamp',
-                value: '{{_ingest.timestamp}}',
-              },
-            },
-          ],
-          on_failure: [
-            {
-              set: {
-                field: 'error.message',
-                value: '{{ _ingest.on_failure_message }}',
-              },
-            },
-          ],
-        });
+        const ingest = getPipelineById(pipelineId);
```

Comment:
I would suggest changing the design here.
You are creating a strong coupling between a utility function like `createPipelineIfNotExists` and the actual pipelines.
A robust design here would be, to introduce a new parameter: `pipelineConf`/`pipelineDesc`/ `pipelineSpec` that describes the pipeline. so if it doesn't exists, you will create it

Reply:
I understand why what you suggest is more robust.

I wanted to keep the changes as minimal as I didn't want to change the function signature. This is why I used a dictionary like a resolver for the different ids.

If you find it important I can change it